### PR TITLE
Add Hubble keyboard

### PR DESCRIPTION
**Pull request by ch3rny**
## Description

Add Hubble PCB to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hubble/hubble.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "hubble.h"
```
Comment (drashna) [inline keyboards/hubble/hubble.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/hubble/hubble.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#pragma once
```
Comment (drashna) [inline keyboards/hubble/hubble.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Also, considering that the info.json has matrix information, this file (along with the hubble.c file) can and should be removed, completely.

File: keyboards/hubble/info.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["F5", "F6", "B4", "E6", "D7", "C6", "D4", "D0"],
+        "rows": ["D1", "F4", "F7", "B5", "B1", "B3", "B6", "B2"]
+    },
+    "processor": "atmega32u4",
+    "url": "",
+    "usb": {
+        "vid": "0x1234",
```
Comment (drashna) [inline keyboards/hubble/info.json:22]: This vendor ID is in use, and needs to be changed. 

https://www.the-sz.com/products/usbid/index.php?v=0x1234&p=&n=

The further pass 0x4000, the better.

File: keyboards/hubble/info.json [hunk 4]
```diff
@@ -0,0 +1,90 @@
+{
+    "manufacturer": "Anton Chernenko",
+    "keyboard_name": "hubble",
+    "maintainer": "ch3rny",
+    "bootloader": "caterina",
```
Comment (drashna) [inline keyboards/hubble/info.json:5]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/hubble/info.json [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["F5", "F6", "B4", "E6", "D7", "C6", "D4", "D0"],
+        "rows": ["D1", "F4", "F7", "B5", "B1", "B3", "B6", "B2"]
+    },
+    "processor": "atmega32u4",
```
Comment (drashna) [inline keyboards/hubble/info.json:19]:
```suggestion
```

File: keyboards/hubble/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/hubble/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/hubble/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,39 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/hubble/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/hubble/readme.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# hubble
+
+![hubble](https://raw.githubusercontent.com/ch3rny/hubble-keyboard/main/assets/main.jpg)
```
Comment (drashna) [inline keyboards/hubble/readme.md:3]: Please rehost on imgur.

File: keyboards/hubble/rules.mk [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes	      # Mouse keys
+EXTRAKEY_ENABLE = yes	      # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+RGBLIGHT_ENABLE = yes       # Enable RGB underglow
+VELOCIKEY_ENABLE = yes      # RGB Effects speed
```
Comment (drashna) [inline keyboards/hubble/rules.mk:17]:
```suggestion
# This file is intentionally left blank
```

File: keyboards/hubble/info.json [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    "maintainer": "ch3rny",
+    "bootloader": "caterina",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (drashna) [inline keyboards/hubble/info.json:13]:
```suggestion
        "nkro": true,
        "rgblight": true,
        "velocikey": true
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/hubble/hubble.c (hunk 1 above)
Comment (ch3rny) [reply keyboards/hubble/hubble.c]: fixed

File: keyboards/hubble/hubble.h (hunk 2 above)
Comment (ch3rny) [reply keyboards/hubble/hubble.h]: removed

File: keyboards/hubble/info.json (hunk 3 above)
Comment (ch3rny) [reply keyboards/hubble/info.json]: updated

File: keyboards/hubble/info.json (hunk 4 above)
Comment (ch3rny) [reply keyboards/hubble/info.json]: fixed

File: keyboards/hubble/info.json (hunk 5 above)
Comment (ch3rny) [reply keyboards/hubble/info.json]: fixed

File: keyboards/hubble/keymaps/default/keymap.c (hunk 6 above)
Comment (ch3rny) [reply keyboards/hubble/keymaps/default/keymap.c:4]: fixed

File: keyboards/hubble/keymaps/via/keymap.c (hunk 7 above)
Comment (ch3rny) [reply keyboards/hubble/keymaps/via/keymap.c:4]: fixed

File: keyboards/hubble/readme.md (hunk 8 above)
Comment (ch3rny) [reply keyboards/hubble/readme.md]: updated

File: keyboards/hubble/rules.mk (hunk 9 above)
Comment (ch3rny) [reply keyboards/hubble/rules.mk]: fixed

File: keyboards/hubble/info.json (hunk 10 above)
Comment (ch3rny) [reply keyboards/hubble/info.json]: fixed

File: keyboards/hubble/config.h [hunk 11]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 Anton Chernenko (@ch3rny)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* key matrix */
+#define MATRIX_ROWS 8
+#define MATRIX_COLS 8
```
Comment (zvecr) [inline keyboards/hubble/config.h:10]:
```suggestion
```

File: keyboards/hubble/readme.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
+# hubble
+
+![hubble](https://i.imgur.com/buTSdUq.jpg)
+
+*A short description of the keyboard/project*
```
Comment (zvecr) [inline keyboards/hubble/readme.md:5]: Needs filling out, or line removing

**Review by zvecr — CHANGES_REQUESTED** [review]
Also,
`keyboards/hubble/rules.mk` should match `data/templates/keyboard/rules.mk`

File: keyboards/hubble/config.h (hunk 11 above)
Comment (ch3rny) [reply keyboards/hubble/config.h]: fixed

File: keyboards/hubble/readme.md (hunk 12 above)
Comment (ch3rny) [reply keyboards/hubble/readme.md]: updated

File: keyboards/hubble/readme.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
+# hubble
+
+![hubble](https://i.imgur.com/buTSdUq.jpg)
```
Comment (drashna) [inline keyboards/hubble/readme.md:3]:
```suggestion
![hubble](https://i.imgur.com/buTSdUqh.jpg)
```

Comment (ch3rny) [reply keyboards/hubble/readme.md]: updated

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**